### PR TITLE
Add emicklei/go-restful instrumentation

### LINKED ISSUE #79: Add instrumentation for go-restful
https://github.com/emicklei/go-restful

## Tasks

- [x] code complete with tests passing
- [x] `Dockerfile` with example app showing instrumentation
- [x] documentation updated
- [x] instrumentation added to the opentelemetry [registry](https://opentelemetry.io/registry/)

## Prior Art

- https://github.com/opentracing-contrib/go-restful
- https://github.com/DataDog/dd-trace-go/tree/v1/contrib/emicklei/go-restful

### PR DESCRIPTION
**Pull request by evantorrie**
fixes #79 

A filter may be added at any one of the following:
   * container scope
   * webservice scope
   * route scope

Includes a docker-compose example with instructions.

### CODE REVIEW

File: instrumentation/emicklei/go-restful/example/server.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+var tracer oteltrace.Tracer
+
+type UserResource struct{}
+
+func (u UserResource) WebService() *restful.WebService {
+	ws := &restful.WebService{}
+
+	ws.
+		Path("/users").
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply instrumentation/emicklei/go-restful/example/server.go:40]: Or, more specifically the one above.

[the remark this answers is not in the document]
Comment (evantorrie) [reply instrumentation/emicklei/go-restful/example/server.go:40]: I was following the style of the go-restful [example from here](https://github.com/emicklei/go-restful#example), but I have no objection to changing it.

**Comment by evantorrie** [comment]
> Did we want to move this to `github.com/emicklei/go-restful`? I know we had talked about this for the other instrumentation that is missing it as well. Maybe a subsequent PR to move everything?

Yes, I planned a subsequent PR to move everything from github.com under a github.com directory.

File: instrumentation/emicklei/go-restful/restful.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	tracerName    = "go.opentelemetry.io/contrib/instrumentation/emicklei/go-restful"
+	tracerVersion = "1.0"
+)
+
+// OtelFilter returns a restful.FilterFunction which will trace an incoming request.
+//
+// The service parameter should describe the name of the (virtual) server handling
+// the request.  Options can be applied to configure the tracer and propagators
+// used for this filter.
+func OtelFilter(service string, opts ...Option) restful.FilterFunction {
```
[the remark this answers is not in the document]
Comment (evantorrie) [reply instrumentation/emicklei/go-restful/restful.go]: I changed it to `OTelFilter` here and in all uses.

File: instrumentation/emicklei/go-restful/example/server.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrAlias) [reply instrumentation/emicklei/go-restful/example/server.go:40]: Oh, interesting. It's weird that they chose to break at a different point on the next line. :shrug: 

Like I said, total nit, but if your into changing I'm all about consistency.

[the remark this answers is not in the document]
Comment (evantorrie) [reply instrumentation/emicklei/go-restful/example/server.go]: Fixed in 114c1fc

**Pull request merged**